github/apps.go

```diff
@@ -38,3 +89,77 @@ func (s *AppsService) ListInstallations(ctx context.Context, opt *ListOptions) (
 
 	return i, resp, nil
 }
+
+// GetInstallation returns the specified installation.
+//
+// GitHub API docs: https://developer.github.com/v3/apps/#get-a-single-installation
+func (s *AppsService) GetInstallation(ctx context.Context, id int) (*Installation, *Response, error) {
+	u := fmt.Sprintf("app/installations/%v", id)
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO: remove custom Accept header when this API fully launches.
+	req.Header.Set("Accept", mediaTypeIntegrationPreview)
+
+	i := new(Installation)
+	resp, err := s.client.Do(ctx, req, i)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return i, resp, nil
+}
+
+// ListUserInstallations lists installations that are accessible to the authenticated user.
+//
+// GitHub API docs: https://developer.github.com/v3/apps/#list-installations-for-user
+func (s *AppsService) ListUserInstallations(ctx context.Context, opt *ListOptions) ([]*Installation, *Response, error) {
+	u, err := addOptions("user/installations", opt)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO: remove custom Accept header when this API fully launches.
+	req.Header.Set("Accept", mediaTypeIntegrationPreview)
+
+	var i struct {
```

Comment:
@shurcooL - what do you think about making this a new `type ... struct` and include the `TotalCount` field, then returning a pointer to the struct in the return value of this method?

I'm asking because it seems more consistent with the rest of the package and also allows GitHub to add/tweak their response and we could more flexibly support it however they decide to tweak it.

Reply:
I agree, I think we should have this method return a pointer to a struct that corresponds to the JSON response in the documentation.

Returning `[]*Installation` would've made sense if the GitHub response was a JSON array of installation objects, but it's a JSON object containing `installations` field with the array.

The only question is what to name it, if we make it exported (I think we have to, don't we?). Perhaps `ListUserInstallationsResponse`?

Reply:
Then again, it's unfortunate it's quite different from the existing `ListInstallations` method (https://developer.github.com/v3/apps/#find-installations) which _does_ return a JSON array of installations...

I can't help but think GitHub might've made the API inconsistent unintentionally and might want to change it while it's still in preview, so it'd be worth shooting them an email about it.

Reply:
@gmlewis @shurcooL I did this in my original contribution, but later removed it because I didn't see this pattern being used in other places in `go-github` where the same type of JSON structure was being handled.

Reply:
> I didn't see this pattern being used in other places in `go-github` where the same type of JSON structure was being handled.

Mind sharing examples of that? It would be helpful.

Reply:
@shurcooL: Same file. This is currently in `master`: https://github.com/google/go-github/blob/master/github/apps_installation.go#L40

But as you can see in the GitHub API docs, the response actually has the `total_count` field.

Reply:
Thanks.

@gmlewis What do you think about this finding?

Reply:
OK, looks like we do this elsewhere... I suppose that if the GitHub API starts expanding these return values to more than just `total_count` that we can then revisit making a named structure.
So I'm fine with proceeding as written.
Thanks, @mandrean and @shurcooL!